Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,122 @@ static void Main(string[] args)
+        {
             var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
             Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
             Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
-            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+            Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
             Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
-            Console.WriteLine(result);
+            Console.WriteLine(result);
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+            var length1 = protein.Length;
+            var length2 = dietPlans.Length;
+
+            int[] array = new int[length2];
+            int[] calorie = new int[length1];
+
+            // Calculate calorie for each element
+            for(var i=0;i<length1;i++){
+                calorie[i]= (protein[i]*5 + carbs[i]*5 + fat[i]*9);
+            }
+         
+
+            for(var i=0;i<length2;i++){
+                var stringLen = dietPlans[i].Length;
+
+                //Use a visit array which will keep track of duplicate element values
+                int[] visit  = new int[length1];
+
+                for(int j=0;j<length1;j++)
+                    visit[j]=1;
+
+                for(int j=0;j<stringLen;j++){
+                    
+                    //Update visit array according to duplicacy in nutrients
+
+                    if(dietPlans[i][j]=='p'){
+                        MinCheck(protein,visit);
+                    }
+                    if(dietPlans[i][j]=='P'){
+                        MaxCheck(protein,visit);
+                    }
+                    if(dietPlans[i][j]=='c'){
+                        MinCheck(carbs,visit);
+                    }
+                    if(dietPlans[i][j]=='C'){
+                        MaxCheck(carbs,visit);
+                    }
+                    if(dietPlans[i][j]=='f'){
+                        MinCheck(fat,visit);
+                    }
+                    if(dietPlans[i][j]=='F'){
+                        MaxCheck(fat,visit);
+                    }
+                    if(dietPlans[i][j]=='t'){
+                        MinCheck(calorie,visit);
+                    }
+                    if(dietPlans[i][j]=='T'){
+                        MaxCheck(calorie,visit);
+                    }
+
+                    //Store index in array 
+                    array[i] =Check(visit);
+
+                }
+
+            }
+            
+            return array;
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static void MinCheck(int[] nutrient,int[] visit){
+            var min = 2000;

```

Comment:
Your 'min' value is fixed in each call, you should use private constant for this so that it will be initialized just once in program lifetime.
Also variable should say what value it is gonna hold. Here it is understood that it is minimum of something. But minimum of what??? carb?? fat???